src/main/resources/db/migration/v0.6.0.1__agr_curation_api.sql

```diff
@@ -0,0 +1,135 @@
+ALTER TABLE reference
+	ADD COLUMN IF NOT EXISTS submittedcrossreference varchar (255),
+	ADD COLUMN IF NOT EXISTS primarycrossreference varchar (255);
+
+ALTER TABLE reference
+	ADD CONSTRAINT reference_submittedcrossreference_uk UNIQUE (submittedcrossreference);
+
+ALTER TABLE reference_aud
+	ADD COLUMN IF NOT EXISTS submittedcrossreference varchar (255),
+	ADD COLUMN IF NOT EXISTS primarycrossreference varchar (255);
+	
+UPDATE reference SET submittedcrossreference = curie;
+
+UPDATE reference SET primarycrossreference = curie;
+
+CREATE TABLE IF NOT EXISTS reference_secondarycrossreferences (
+	reference_submittedcrossreference varchar (255),
+	secondarycrossreferences varchar (255)
+	);
+	
+CREATE TABLE IF NOT EXISTS reference_secondarycrossreferences_aud (
+	rev integer,
+	reference_submittedcrossreference varchar (255),
+	secondarycrossreferences varchar (255),
+	revtype smallint
+	);
+	
+ALTER TABLE reference_secondarycrossreferences
+	ADD CONSTRAINT reference_secondarycrossreferences_submittedcrossreference_fk
+	FOREIGN KEY (reference_submittedcrossreference) REFERENCES reference(submittedcrossreference);
+	
+CREATE TABLE IF NOT EXISTS note_reference (
+	note_id bigint,
+	references_curie varchar (255)
+	);
+
+CREATE TABLE IF NOT EXISTS note_reference_aud (
```

Comment:
Also can we get away from using "IF NOT EXISTS" either the table exists or it doesn't?

Reply:
We have to do this because of the integration tests.  I think it's something to do with tables that didn't exist in the base schema.  They weren't explicitly created in the initial migrations (because we were letting Hibernate do that), but now we need to create it.  Outside of the integration tests, the table already exists.

Reply:
Or in most of these cases I think it's because we're still allowing Hibernate to create tables in dev mode.  If you do development and then write the migration file, those tables have already been created once the migration runs.  I don't know, this migration stuff makes my head hurt!  I think in some cases I've added 'IF NOT EXISTS' due to failing migration scripts, but I've possibly been overly liberal with them.  Maybe we should always revert our local database to the same state as alpha before running the migration script locally to avoid this, but does it really matter if we leave some unnecessary 'IF NOT EXISTS's in?

Reply:
In my previous commit with the migration change I went through and made sure that ALL tables were created in the base schema, so that either via hibernate or via migration the DB was the same.

Well the issue I see is that you never know if the statement actual ran or not. Integration tests should not fail due to missing tables as I made sure in my commit a few days ago that everything is in the base schema. And yes I agree that we should be taking the schema as in on Alpha, and then running the migrations against it even if the current code has us creating tables via hibernate.